src/adapter/src/coord/command_handler.rs

```diff
@@ -186,13 +186,14 @@ impl Coordinator {
 
     async fn handle_startup(
         &mut self,
-        session: Session,
+        mut session: Session,
         cancel_tx: Arc<watch::Sender<Canceled>>,
         tx: oneshot::Sender<Response<StartupResponse>>,
     ) {
+        // Lookup is done with the system session because the current session has no role set.
         if self
             .catalog
-            .for_session(&session)
+            .for_system_session()
```

Comment:
Instead of calling for_system_session twice can you save it for reuse?

Reply:
We can't because `for_sessionless_user` holds on to an immutable borrow of `self`, which then errors when we try and call `sequence_create_role` which requires a mutable borrow of `self`.